pkg/parser/phc2sys_parser.go

```diff
@@ -0,0 +1,206 @@
+		//  0             0       1   2
+		//ptp4l.0.config master rms   53 max   74 freq -16642 +/-  40 delay  1089 +/-  20
+	}
+
+	iface = fields[1]
+
+	ptpOffset, err = strconv.ParseFloat(fields[3], 64)
+	if err != nil {
+		glog.Errorf("%s failed to parse offset from the output %s error %v", PTP4L, fields[3], err)
+	}
+
+	maxPtpOffset, err = strconv.ParseFloat(fields[5], 64)
+	if err != nil {
+		glog.Errorf("%s failed to parse max offset from the output %s error %v", PTP4L, fields[5], err)
+	}
+
+	frequencyAdjustment, err = strconv.ParseFloat(fields[7], 64)
+	if err != nil {
+		glog.Errorf("%s failed to parse frequency adjustment output %s error %v", PTP4L, fields[7], err)
+	}
+
+	if len(fields) >= 11 {
+		delay, err = strconv.ParseFloat(fields[11], 64)
+		if err != nil {
+			glog.Errorf("%s failed to parse delay from the output %s error %v", PTP4L, fields[11], err)
+		}
+	} else {
+		// If there is no delay from master this mean we are out of sync
+		glog.Warningf("no delay from master process %s out of sync", PTP4L)
+	}
+
+	offsetSource := master
+
+	return nil, &Metrics{
+		Iface:      iface,
+		Offset:     ptpOffset,
+		MaxOffset:  maxPtpOffset,
+		FreqAdj:    frequencyAdjustment,
+		Delay:      delay,
+		ClockState: clockState,
+		Source:     offsetSource,
+	}
+
+}
+
+func extractRegularPhc2Sys(configName, processName, output string, ifaces config.IFaces) (error, *Metrics) {
+	indx := strings.Index(output, offset)
+	if indx < 0 {
+		return nil, nil
+	}
+	output = normalizeLine(output)
+
+	index := strings.Index(output, configName)
+	if index == -1 {
+		return nil, nil
+	}
+
+	output = output[index:]
+	fields := strings.Fields(output)
+	var err error
+	var iface, clockState string
+	var ptpOffset, maxPtpOffset, frequencyAdjustment, delay float64
+
+	//       0         1      2          3     4   5    6          7     8
+	// ptp4l.0.config master offset   -2162130 s2 freq +22451884  delay 374976
+
+	if len(fields) < 7 {
+		return nil, nil
+	}
+	//       0         1      2          3    4   5       6     7     8
+	//ptp4l.0.config master offset       4    s2  freq   -3964 path delay  91
+
+	if len(fields) < 7 {
+		err = fmt.Errorf("%s failed to parse output %s: unexpected number of fields", processName, output)
+		return nil, nil
+	}
+
+	if fields[2] != offset {
+		err = fmt.Errorf("%s failed to parse offset from the output %s error %s", processName, fields[1], "offset is not in right order")
+		return nil, nil
+	}
+
+	iface = fields[1]
+	if iface != clockRealTime {
+		return nil, nil // ignore other interfaces offsets
+	}
+
+	ptpOffset, e := strconv.ParseFloat(fields[3], 64)
+	if e != nil {
+		err = fmt.Errorf("%s failed to parse offset from the output %s error %v", processName, fields[1], err)
+		return nil, nil
+	}
+
+	maxPtpOffset, err = strconv.ParseFloat(fields[3], 64)
+	if err != nil {
+		err = fmt.Errorf("%s failed to parse max offset from the output %s error %v", processName, fields[1], err)
+		return nil, nil
+	}
+
+	switch fields[4] {
```

Comment:
Should this be its own function for reuse as seems to be a common thing for linuxptp